### PR TITLE
[serve] Make `serve.shutdown()` shut down remote Serve applications

### LINKED ISSUE #22254: [Bug] Serve shutdown does not work for detached serve instance
### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/ray/issues) and found no similar issues.

### Ray Component

Ray Serve

### What happened + What you expected to happen

Trying to shutdown a detached ray serve instance using `serve.shutdown` does not work.

### Versions / Dependencies

ray==1.10.0

python=3.8.0

NAME="Ubuntu"
VERSION="18.04.6 LTS (Bionic Beaver)"

### Reproduction script

```
import ray
ray.init(address="ray://example-cluster-ray-head.ray:10001", namespace="x")
from ray import serve
serve.shutdown()
```

### Anything else

The problem is resolved if another serve API is called before shutdown

```
import ray
ray.init(address="ray://example-cluster-ray-head.ray:10001", namespace="x")
from ray import serve
serve.list_deployments()
serve.shutdown()
```

I think this if clause is unnecessary and it is what is causing the bug - https://github.com/ray-project/ray/blob/master/python/ray/serve/api.py#L703

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by shrekris-anyscale**
## Why are these changes needed?

`serve.shutdown()` does not do anything if there is no cached handle to a Serve controller. If a user connects to a remote Ray cluster without running `serve.start()`, they cannot use `serve.shutdown()` to shut down the remote Serve application.

E.g. this does nothing, even if a Serve controller is running at `"cluster_address"` in namespace `"x"`:

```python
import ray
from ray import serve

ray.init(address="cluster_address", namespace="x")
serve.shutdown()
```

This change makes `serve.shutdown()` first try to connect to a Serve controller before attempting to shut down. This allows a user to directly shut down a running Serve instance without potentially having to start a Serve instance with `serve.start()` first.

## Related issue number

Closes #22254.

## Checks

- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [X] Unit tests
     - New unit tests are added to `test_standalone2.py`.

### CODE REVIEW

File: python/ray/serve/api.py [hunk 1]
```diff
@@ -857,10 +865,17 @@ def shutdown() -> None:
...
     """
-    if _global_client is None:
+
+    try:
+        client = internal_get_global_client()
+    except RayServeException:
+        logger.info(
+            "Nothing to shut down. There's no Serve application "
+            "running on this Ray cluster."
+        )
```
Comment (edoakes) [inline python/ray/serve/api.py:875]: this would be a breaking change, can we instead just return silently if there's nothing to shut down?

**Review by edoakes — CHANGES_REQUESTED** [review]

File: python/ray/serve/api.py (hunk 1 above)
Comment (edoakes) [reply python/ray/serve/api.py:875]: sorry, disregard me here...

**Review by edoakes — APPROVED** [review]

**Comment by shrekris-anyscale** [comment]
@edoakes It is a real failure– Windows has some issues when trying to run temp files. I can either skip the test on Windows or refactor it to pass. I'll look into this.

**Comment by shrekris-anyscale** [comment]
@edoakes I removed the context manager from the test per https://stackoverflow.com/a/15590253, and it looks like `test_standalone2.py` is passing in Windows and Linux now.

**Review by edoakes — APPROVED** [review]

**Comment by edoakes** [comment]
@shrekris-anyscale can we update the `DeploymentNode` docstring to say "deployment graph" instead of "multi-deployment applicaton" ?

**Comment by edoakes** [comment]
Sigh... wrong PR. Clearly reviewing will distracted today...

**Pull request merged**